strictdoc/commands/manage_autouid_command.py

```diff
@@ -91,3 +109,69 @@ def execute(
                 document_meta.input_doc_full_path, "w", encoding="utf8"
             ) as output_file:
                 output_file.write(document_content)
+
+        for (
+            trace_info_
+        ) in traceability_index.get_file_traceability_index().trace_infos:
+            assert trace_info_.source_file is not None
+            # FIXME: These conditions for skipping the writes may be insufficient.
+            if (
+                not trace_info_.source_file.in_doctree_source_file_rel_path_posix.endswith(
+                    ".c"
+                )
+                or not trace_info_.source_file.is_referenced
+            ):
+                continue
+            with open(trace_info_.source_file.full_path, "rb") as source_file_:
+                file_bytes = source_file_.read()
+
+            rewrites = {}
+            for source_node_ in trace_info_.source_nodes:
+                function = source_node_.function
+                if function is None:
+                    continue
+
+                # FILE_PATH: The file the code resides in, relative to the root of the project repository.
+                file_path = bytes(
+                    trace_info_.source_file.in_doctree_source_file_rel_path_posix,
+                    encoding="utf8",
+                )
+
+                # INSTANCE:	The requirement template instance, minus tags with hash strings.
+                instance_bytes = bytearray()
+                for field_name_, field_value_ in source_node_.fields.items():
+                    if field_name_ in ("SPDX-Req-ID", "SPDX-Req-HKey"):
+                        continue
+                    instance_bytes += bytes(field_value_, encoding="utf8")
+
+                # CODE: The code that the SPDX-Req applies to.
+                code = file_bytes[
+                    function.code_byte_range.start : function.code_byte_range.end
+                ]
+
+                # This is important for Windows. Otherwise, the hash key will be calculated incorrectly.
+                code = code.replace(b"\r\n", b"\n")
+
+                hash_spdx_id = bytes(get_random_sha256(), encoding="utf8")
+                hash_spdx_hash = generate_code_hash(
+                    project=bytes(
+                        project_config.project_title, encoding="utf8"
+                    ),
+                    file_path=file_path,
+                    instance=bytes(instance_bytes),
+                    code=code,
+                )
+                patched_node = MarkerWriter().write(
```

Comment:
At least SPDX-Req-ID is meant to be remapped to MID. So we have to not only update the source code with the new calculated value, but also all links in the graph database. And also all links in the search index. This is not done yet, is it?